bim2sim/elements/mapping/attribute.py

```diff
@@ -294,51 +294,49 @@ def request(self, bind, external_decision=None):
             external_decision: Decision to use instead of default decision
         """
 
-        # read current value and status
-        value, status, data_source = self._inner_get(bind)
+        # Read current value, status, and data source
+        value, status, _ = self._inner_get(bind)
 
-        if value is None:
-            if status == Attribute.STATUS_NOT_AVAILABLE:
-                _decision = self.get_dependency_decisions(
-                    bind, external_decision)
-                return _decision
-        elif isinstance(value, list):
-            if not all(value):
-                _decision = self.get_dependency_decisions(bind,
-                                                          external_decision)
-                return _decision
-            return
-        elif isinstance(value, Decision):
-            # already a decision stored in value
+        # Case 1: Value is None and status is STATUS_NOT_AVAILABLE
+        if value is None and status == Attribute.STATUS_NOT_AVAILABLE:
+            return self.get_dependency_decisions(bind, external_decision)
+
+        # Case 2: Value is a list and not all elements are truthy
+        if isinstance(value, list) and not all(value):
+            return self.get_dependency_decisions(bind, external_decision)
+
+        # Case 3: Value is already a Decision instance
+        if isinstance(value, Decision):
             return value
-        else:
-            # already requested or available
-            return
+
+        # Case 4: Value is available or already requested (no action needed)
+        return
 
     def get_dependency_decisions(self, bind, external_decision=None):
         """Get dependency decisions"""
         if self.functions is not None:
-            if self.dependant_elements:
-                _decision = {}
+            if not self.dependant_elements:
+                _decision = external_decision or self.create_decision(
```

Comment:
line break is not needed here
```suggestion
_decision = external_decision or self.create_decision(bind)
```